docs_app/src/app/custom-elements/demo/events-advanced/events-advanced.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'aio-events-advanced',
+  templateUrl: './events-advanced.component.html',
+  styles: []
+})
+export class EventsAdvnacedComponent implements OnInit {
+
+  countries;
+  statesForSelectedCountry;
+
+  data = [{
+    id: 1,
+    name : 'India',
+    states : [{
+      id: 1,
+      name: 'Uttar Pradesh'
+    },{
+      id: 2,
+      name: 'Bihar'
+    },{
+      id: 3,
+      name: 'Rajasthan'
+    }]
+  },{
+    id: 2,
+    name: 'USA',
+    states: [{
+      id : 1,
+      name: 'New York'
+    },{
+      id: 2,
+      name: 'Arizona'
+    }],
+  }];
+
+  constructor() { }
+
+  ngOnInit() {
+    this.countries = this.data;
+  }
+
+  onCountrySelection(selectedCountry) {
+    this.statesForSelectedCountry = this.data.find(country => country.name === selectedCountry.name).states;
```

Comment:
What if some value gets selected and `country` dropdown options gets cleared. I believe it should clear state dropdown selected value.